account_commission_listing_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    event_ids = fields.Many2many(
+        "event.event", compute="_compute_event_ids", string="Events", store=True
+    )
+    number_of_items = fields.Integer(compute="_compute_number_of_items", store=True)
+
+    @api.depends("line_ids")
+    def _compute_event_ids(self):
+        for settlement in self:
+            events = settlement.line_ids.product_id.event_ids
+            settlement.event_ids = events
+
+    @api.depends("event_ids", "agent_id")
+    def _compute_number_of_items(self):
+        for settlement in self:
+            event_ids = settlement.event_ids.ids
+            if not event_ids:
+                settlement.number_of_items = 0
+                continue
+            products = (
+                self.env["product.product"]
+                .search(
+                    [
+                        ("seller_id", "=", settlement.agent_id.id),
+                        "|",
+                        ("settlement_id", "=", False),
+                        ("settlement_id", "=", settlement.id),
+                    ]
+                )
+                .filtered(
+                    lambda x: x.event_ids and set(x.event_ids.ids).issubset(event_ids)
```

Comment:
```suggestion
                    lambda x:  event for event in x.event_ids if event in event_ids
```
I think we should not use `issubset()` any more since we already apply a filter by settlement_id.